peer-pod-controller/api/v1alpha1/peerpod_types.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright Confidential Containers Contributors
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// PeerPodSpec defines the desired state of PeerPod
+type PeerPodSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Foo is an example field of PeerPod. Edit peerpod_types.go to remove/update
+	CloudProvider string `json:"cloudProvider,omitempty"`
```

Comment:
Not really, at first i used it as it was not very simple to pass parameters at build time using the operator-sdk.
I'm still not sure how to do it but it indeed will be useful if we'll support multi cloud, and generally it just make sense to have context about the which provider the InstanceID refers to

Reply:
OK. If we want to support multiple cloud providers, there is also possibility to support multiple accounts of a single cloud provider. So, we need a label to represents a pair of cloud provider name and account identifier.

A possible simple way to support such situation is to run multiple controllers, each of which is responsible only for a single cloud provider and a single account.  A controller should have a startup option to specify a value to be used for the label. The default value can be the same string of a cloud provider, since a single account is usually used for a single cloud provider.

Each controller determines which PeerPod objects it should handle by filtering the objects with the label.

Reply:
I think is OK to keep this field. We will come back when we support multiple cloud providers.